### PR TITLE
release: version v10.4.0

### PR DESCRIPTION
**Pull request by polarathene**
# Description

This PR marks the feature freeze phase for the `v10.4.0` release.

The current `:edge` image will be tested for roughly one week and if no bugs get reported, `v10.4.0` will be released. During this testing period, only PRs that do not change functionality will be merged.

**If you want to support us, give the latest `:edge` image a try and report any issues you encounter back to this PR.**

**Feature freeze is in place until _24 Dec 2021_**. Documentation updates may be merged.

## Type of change

- [x] Release

## Release-Checklist:

- [x] Update CHANGELOG.md
- [x] Update version in VERSION
- [x] Draft release

### CODE REVIEW

File: CHANGELOG.md
Comment (casperklein) [inline CHANGELOG.md:11]: Afaik, only the debian package of F2B is affected by this. We are using the official '0.11.2' release, which was released before the mentioned patch. There was no new official release since then.

File: CHANGELOG.md
Comment (polarathene) [inline CHANGELOG.md:12]: Removing the `fail2ban` mention as @casperklein [points out we use upstream directly, not the debian package](https://github.com/docker-mailserver/docker-mailserver/pull/2328/files#r771818066), and that apparently makes us unaffected? (We do use the `bsd-mailx` debian package though)

```suggestion
  - `yescrypt` is now supported upstream as a password hash algorithm, `docker-mailserver` continues to use `SHA512-CRYPT` (_[more information](https://github.com/docker-mailserver/docker-mailserver/pull/2116#issuecomment-955800544)_).
```

I'll delay applying the change in case anyone wants to discuss further before I drop the warning from the changelog.

Comment (georglauterbach) [reply CHANGELOG.md:12]: I'd remove it as well as I think @casperklein is correct :)

**Review by georglauterbach — APPROVED** [review]

**Comment by casperklein** [comment]
I recommend approving such kind of PRs only after successful testing.

**Comment by georglauterbach** [comment]
You're right.

**Comment by georglauterbach** [comment]
"Don't count your chickens before they are hatched." @casperklein you we're right. I just checked the logs again and found something... (but nothing serious). Enough to block this PR though until #2336 is merged. Sorry :D

**Review by georglauterbach — CHANGES_REQUESTED** [review]
Just blocking this here so no accidental merged until #2336 is merged.

**Comment by georglauterbach** [comment]
Nice! Will accept when we're about to release :) (learned my lesson)

**Comment by casperklein** [comment]
What about #2330. As you labeled it for 10.4.0, it's fine to merge?

**Comment by georglauterbach** [comment]
> What about #2330. As you labeled it for 10.4.0, it's fine to merge?

Yes, I think this can / should go into `v10.4` too.

**Comment by georglauterbach** [comment]
@polarathene are you going to release this on the 24th?:)

**Comment by polarathene** [comment]
> @polarathene are you going to release this on the 24th?:)

Sure, is there a specific timezone? Otherwise I'd be happy to release around this time tomorrow (which is 24th in NZ)

**Comment by georglauterbach** [comment]
> > @polarathene are you going to release this on the 24th?:)
> 
> Sure, is there a specific timezone? Otherwise I'd be happy to release around this time tomorrow (which is 24th in NZ)

Nice!
I think time zone does not matter, just go ahead :)

**Review by georglauterbach — APPROVED** [review]

**Review by wernerfred — APPROVED** [review]
Approving as @polarathene cannot approve his own PR

**Pull request merged**
